### PR TITLE
CRF Additions

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
- Added a step and view controller for selecting an image/video using an image picker controller.
- Fixed a few Bridge refactor changes to work for existing CRF module.
- Implemented CSV file support.
- Implemented result archiving.

### CODE REVIEW

File: ResearchSuite/ResearchSuite/RSDTaskPath.swift [hunk 1]
```diff
@@ -278,23 +282,80 @@ public final class RSDTaskPath : NSObject, NSCopying {
...
+                print("Error removing output directory: \(error.localizedDescription)")
+                debugPrint("\tat: \(outputDirectory)")
+            }
+            completion?()
+        }
+    }
+    
+    /// Build an archive from the task result.
+    ///
+    /// This method will recursive through the task result and pull out data for archiving using the given
```
Comment (Erin-Mounts) [inline ResearchSuite/ResearchSuite/RSDTaskPath.swift]: recursive --> recurse

File: ResearchSuite/ResearchSuite/en.lproj/ResearchSuite.strings [hunk 2]
```diff
@@ -105,3 +105,10 @@
 "MEDICATION_DOSAGE_PLACEHOLDER" = "e.g. 10/100 mg";    // Placeholder text for medication dosage.
 "MEDICATION_DAYS_OF_WEEK_TITLE_%1$@_at_%2$@" = "Which days do you take %1$@ at %2$@?"; // ex. "Advil at 7:00 AM"
 "MEDICATION_REVIEW_TITLE" = "Review medications";
+
+/* Image Picker */
+"CAMERA_NOT_AVAILABLE" = "This device does not support using the camera to capture images.";
+"PHOTO_LIBRARY_PERMISSION_DENIED" = "You have not given this app permission to use the photo library. You can enable access in the Privacy Settings.";
+"PHOTO_LIBRARY_PERMISSION_RESTRICTED" = "Access to the photo library is restricted. You must allow permission within 'Settings -> General -> Restrictions'";
+"CAMERA_PERMISSION_DENIED" = "You have not given this app permission to use the photo library. You can enable access in the Privacy Settings.";
```
Comment (Erin-Mounts) [inline ResearchSuite/ResearchSuite/en.lproj/ResearchSuite.strings]: photo library --> camera

File: ResearchSuite/ResearchSuite/RSDCommaSeparatedEncodable.swift [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// A special-case encodable that can be encoded to a comma-delimited string.
+///
+/// A csv-formatted file is a smaller format that might be suitable for saving data to a file that will
+/// be parsed into a table, **but** the elements must all conform to single value container encoding
+/// **and** they may not include any strings in the encoded value.
```
[the remark this answers is not in the document]
Comment (syoung-smallwisdom) [reply ResearchSuite/ResearchSuite/RSDCommaSeparatedEncodable.swift]: yes.

File: ResearchSuite/ResearchSuite/RSDImagePickerStep.swift [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// What source should be used for the image selection?
+/// - note: The app is required to include the appropriate permissions in its `Info.plist` file.
```
[the remark this answers is not in the document]
Comment (syoung-smallwisdom) [reply ResearchSuite/ResearchSuite/RSDImagePickerStep.swift]: not sure that makes more sense, but sure?

**Pull request merged**